codegen/smithy-aws-ruby-codegen/src/main/java/software/amazon/smithy/aws/ruby/codegen/AWSIntegration.java

```diff
@@ -116,28 +113,22 @@ metadata service (IMDS) on an EC2 instance.
                 * `~/.aws/credentials` and `~/.aws/config`
                 * EC2/ECS instance profiles.
                   
-                @see AWS::SDK::Core::CREDENTIALS_PROVIDER_CHAIN""";
-
-        Symbol awsCredentialsIdentity = Symbol.builder()
-                .namespace("AWS::SDK::Core::Identities", "::")
-                .name("Credentials").build();
-        String identityProviderChain = " *AWS::SDK::Core::CREDENTIALS_PROVIDER_CHAIN";
-
-        String defaultIdentity = "%s.new(access_key_id: 'stubbed-akid', secret_access_key: 'stubbed-secret')"
-                .formatted(awsCredentialsIdentity);
-        String defaultConfigValue = "cfg[:stub_responses] ? %s.new(proc { %s }) : nil"
-                .formatted(Hearth.IDENTITY_PROVIDER, defaultIdentity);
+                @see AWS::SDK::Core::CredentialsProviderChain
+                """;
 
         ClientConfig identityProviderConfig = ClientConfig.builder()
                 .name("credentials_provider")
                 .documentationRbsAndValidationType(Hearth.IDENTITY_PROVIDER.toString())
                 .documentation(credentialProviderDocumentation)
                 .defaults(ConfigProviderChain.builder()
-                        .dynamicProvider(defaultConfigValue)
-                        .staticProvider(identityProviderChain)
+                        .staticProvider("AWS::SDK::Core::CredentialsProviderChain.new")
```

Comment:
This doesn't look correct to me.  How is stubbed responses handled?  Shouldn't this be a dynamic provider and accept the cfg?  Or do you mean to use the `call` method?